### ISSUE TITLE
[UMP] Application crash when calling Update of ConsentInformation on iOS

### ISSUE THREAD

**Issue by kevinchailloux** [googleads/googleads-mobile-unity#2617]
### [REQUIRED] Step 1: Describe your environment

  * Unity version: 2021.3.21
  * Google Mobile Ads Unity plugin version: 8.0.0
  * Platform: iOS
  * Platform OS version: iOS 15.6
  * Any specific devices issue occurs on: iPhone SE
  * Mediation ad networks used, and their versions: None

#### Steps to reproduce:

- Initialize the Mobile Ads SDK
- Call ConsentInformation.Update()

#### Relevant Code:

```csharp
MobileAds.RaiseAdEventsOnUnityMainThread = true; //tried with and without
MobileAds.Initialize(status => {
    ConsentRequestParameters consentRequestParameters = new ConsentRequestParameters();
    ConsentInformation.Update(consentRequestParameters, OnConsentInfoUpdated);
});
```

The generated error is
```
2023-04-07 10:51:09.378639+0200 ads-kit[94471:4050865]  - <Google>[I-ACS025031] AdMob App ID changed. Original, new: (nil), ca-app-pub-1222650526886905~2139676866
2023-04-07 10:51:09.383132+0200 ads-kit[94471:4050868]  - <Google>[I-ACS023007] Analytics v.100600000 started
2023-04-07 10:51:09.383630+0200 ads-kit[94471:4050868]  - <Google>[I-ACS023008] To enable debug logging set the following application argument: -APMAnalyticsDebugEnabled (see http://goo.gl/RfcP7r)
2023-04-07 10:51:09.384822+0200 ads-kit[94471:4050589] Unbalanced calls to begin/end appearance transitions for <SplashScreenController: 0x1317456b0>.
2023-04-07 10:51:09.423968+0200 ads-kit[94471:4050868]  - <Google>[I-ACS800023] No pending snapshot to activate. SDK name: app_measurement
2023-04-07 10:51:09.486656+0200 ads-kit[94471:4050868]  - <Google>[I-ACS023012] Analytics collection enabled
UnloadTime: 1.540625 ms
Last log before calling consent info update
System.Collections.Generic.List`1:ForEach(Action`1)
MWMKits.AdsManager.Helper.Consent:Update()

libc++abi: terminating with uncaught exception of type Il2CppExceptionWrapper
dyld4 config: DYLD_INSERT_LIBRARIES=/Developer/usr/lib/libMainThreadChecker.dylib:/Developer/Library/PrivateFrameworks/DTDDISupport.framework/libViewDebuggerSupport.dylib
terminating with uncaught exception of type Il2CppExceptionWrapper
(lldb) 
```
Screenshot of my Xcode at the time of the crash :
![Capture d’écran 2023-04-07 à 11 18 21](https://user-images.githubusercontent.com/8959709/230582506-dd9f358b-9eb0-449b-ab7d-173be5f91515.png)

**Comment by NVentimiglia** [comment]
@kevinchailloux 

I have verified the issue and will push for a fix. Thank you.

**Comment by NVentimiglia** [comment]
This should be resolved in the latest release.

**Comment by mancj** [comment]
It is very strange that none of the 16 issues opened here do not contain a solution to this error. I spent a whole day trying to figure out the cause of the error. crash rate dropped to 89% from 100%. a lot of nerves and money spent. 

So if someone is facing this problem, **try putting your** code in `MobileAdsEventExecutor.ExecuteInUpdate`, as described in [this message](https://github.com/googleads/googleads-mobile-unity/issues/2710#issuecomment-1548683969).
I tried 3 versions of library (current actual version is 8.5.2), finally tested successfully on 8.4.1. Hope this will help you.

I searched the whole Internet, but there was no information on this problem anywhere. ExecuteInUplace is not even mentioned in the Google documentation. Google, are you kidding us?
